### PR TITLE
Store morefs in VCH config from vic-machine

### LINKED ISSUE #666: Refactor vic-mache: use mob-id, instead of name in configuration
This issue has dependency on VCH service changes, so put it back to backlog temporarily.

### PR DESCRIPTION
**Pull request by andrewtchin**
Store morefs instead of various other types
Fixes #666

### CODE REVIEW

File: lib/metadata/virtual_container_host.go [hunk 1]
```diff
@@ -63,14 +63,15 @@ type VirtualContainerHostConfigSpec struct {
...
 
 	// Port Layer - storage
-	// Datastore URLs for image stores - the top layer is [0], the bottom layer is [len-1]
-	ImageStores []string `vic:"0.1" scope:"read-only" key:"image_stores"`
+	// Datastore used for the ImageStore
+	// TODO this will change, probably to url.URL, as needed by the port layer
+	ImageStore string `vic:"0.1" scope:"read-only" key:"image_store"`
 	// Permitted datastore URL roots for volumes
-	VolumeLocations []url.URL `vic:"0.1" scope:"read-only" recurse:"depth=0"`
+	VolumeLocations []types.ManagedObjectReference `vic:"0.1" scope:"read-only" recurse:"depth=0"`
```
Comment (emlin) [inline lib/metadata/virtual_container_host.go]: So the image datastore attributes are still plain string.
This is fine, but as we'll move all datastore related attributes to URL eventually, you might not change these definitions to ManagedObjectReference.

File: lib/metadata/virtual_container_host.go [hunk 2]
```diff
@@ -106,6 +107,9 @@ type VirtualContainerHostConfigSpec struct {
 	ResourcePoolPath       string `vic:"0.1" scope:"read-only" key:"resource_pool_path"`
 	ApplianceInventoryPath string `vic:"0.1" scope:"read-only" key:"appliance_path"`
 
+	Datacenter types.ManagedObjectReference `vic:"0.1" scope:"read-only" key:"datacenter"`
+	Cluster    types.ManagedObjectReference `vic:"0.1" scope:"read-only" key:"cluster"`
+
 	// FIXME: remove following attributes after change to launch through tether
 	// Networks represents mapping between nic name and network info object. For example: bridge: vmomi object
 	Networks map[string]*NetworkInfo `vic:"0.1" scope:"read-only" key:"networks"`
```
Comment (emlin) [inline lib/metadata/virtual_container_host.go]: So you didn't add network reference into NetworkInfo?

Comment (andrewtchin) [inline lib/metadata/virtual_container_host.go]: Fixed by adding PortGroupRef

**Comment by emlin** [comment]
LGTM

**Comment by hmahmood** [comment]
lgtm

**Comment by andrewtchin** [comment]
self lgtm

**Pull request merged**